core/scratch_block_comment.js

```diff
@@ -0,0 +1,404 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2011 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Object representing a code comment.
+ * @author kchadha@scratch.mit.edu (Karishma Chadha)
+ */
+'use strict';
+
+goog.provide('Blockly.ScratchBlockComment');
+
+goog.require('Blockly.Comment');
+goog.require('Blockly.ScratchBubble');
+goog.require('Blockly.Events.BlockChange');
+goog.require('Blockly.Events.Ui');
+goog.require('Blockly.Icon');
+goog.require('goog.userAgent');
+goog.require('goog.math.Coordinate');
+
+
+/**
+ * Class for a comment.
+ * @param {!Blockly.Block} block The block associated with this comment.
+ * @param {?number} x Initial x position for comment
+ * @param {?number} y Initial y position for comment
+ * @param {?boolean} minimized Whether or not this comment is minimized
+ * (only the top bar displays), defaults to false
+ * @extends {Blockly.Comment}
+ * @constructor
+ */
+Blockly.ScratchBlockComment = function(block, x, y, minimized) {
+  Blockly.ScratchBlockComment.superClass_.constructor.call(this, block);
+  this.x = x;
+  this.y = y;
+  this.isMinimized_ = minimized || false;
+};
+goog.inherits(Blockly.ScratchBlockComment, Blockly.Comment);
+
+/**
+ * Width of bubble.
+ * @private
+ */
+Blockly.ScratchBlockComment.prototype.width_ = 200;
+
+/**
+ * Height of bubble.
+ * @private
+ */
+Blockly.ScratchBlockComment.prototype.height_ = 200;
+
+/**
+ * Comment Icon Size.
+ */
+Blockly.ScratchBlockComment.prototype.SIZE = 0;
```

Comment:
This probably doesn't need to be on the prototype.  Also, these constants probably don't need to be public.  They are probably private, package, or protected.

Reply:
This is on the prototype because it's overriding the `SIZE` on the prototype of `Blockly.Icon`.